### DISCUSSION TITLE
jextract like tool for Objeck?

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#158] [Ideas]
https://developer.okta.com/blog/2022/04/08/state-of-ffi-java

**Comment by ghost** [comment]
Objeck's current way to call C/C++ is much simpler than JNI and I liked it. But I personally have never used it. When reading code I have real troubles to understand what `(context, 0)`, `(context, 1)`, `(context, 2)`,... and so on is and what it's really get and set. Could you tell me more about it @objeck?

For example:

https://github.com/objeck/objeck-lang/blob/master/core/lib/experimental/gtk3/gtk3.cpp

There is no documents about it, even on lib_api.h there is also has no comments!

**Comment by objeck** [comment]
The ``VMContext& context`` object is used to pass values to and from the VM to C shared libraries. Values are passed in an array. I usually use ``APITools_SetxxxValue(context, 0, value)`` to set return values and ``1-n`` for parameters.
